src/core/flate_stream.js

```diff
@@ -148,6 +149,50 @@ class FlateStream extends DecodeStream {
     this.codeBuf = 0;
   }
 
+  get isAsync() {
+    return true;
+  }
+
+  async getImageData(length, _ignoreColorSpace) {
+    const data = await this.asyncGetBytes();
+    if (!data) {
+      return this.getBytes(length);
+    }
+    return data.subarray(0, length);
+  }
+
+  async asyncGetBytes() {
+    this.str.reset();
+    const bytes = this.str.getBytes();
+
+    try {
+      const { readable, writable } = new DecompressionStream("deflate");
+      const writer = writable.getWriter();
+      writer.write(bytes);
+      writer.close();
+
+      const chunks = [];
+      let totalLength = 0;
+
+      for await (const chunk of readable) {
+        chunks.push(chunk);
+        totalLength += chunk.byteLength;
+      }
+      const data = new Uint8Array(totalLength);
+      let offset = 0;
+      for (const chunk of chunks) {
+        data.set(chunk, offset);
+        offset += chunk.byteLength;
+      }
+
+      return data;
+    } catch {
+      this.str = new Stream(bytes, 2, bytes.length, this.str.dict);
```

Comment:
Sorry, but why do we need to replace `this.str` here? Assuming it's necessary, please add a comment.
Also, where does the magic number `2` come from here?